src/test/java/com/purbon/kafka/topology/TopologySerdesTest.java

```diff
@@ -191,10 +191,14 @@ public void testCoreElementsProcessing() throws IOException, URISyntaxException
     Topology topology = parser.deserialise(Paths.get(topologyDescriptor.toURI()).toFile());
 
     Project project = topology.getProjects().get(0);
-    assertThat(project.getProducers()).hasSize(1);
+    assertThat(project.getProducers()).hasSize(3);
     assertThat(project.getConsumers()).hasSize(2);
     assertThat(project.getStreams()).hasSize(1);
     assertThat(project.getConnectors()).hasSize(2);
+
+    assertThat(project.getProducers().get(0).idempotenceValue()).isEqualTo(false);
+    assertThat(project.getProducers().get(1).transactionIdString()).isEqualTo("1234");
+    assertThat(project.getProducers().get(2).idempotenceValue()).isEqualTo(true);
```

Comment:
Just som minor code details; maybe just use methods on OptionalAssert? Like:
```
    assertThat(project.getProducers().get(0).getIdempotence()).isEmpty();
```

Could then get rid of the test-only methods for getting values on Producer. :-)

Reply:
i like that one! thanks! methods are not that much only for test! but I agree your point! thanks